Bottom: 799d9bfcd5b6545c40e5e19c7b50907bd25ed616
Top:    2eb494d99a798d8027fb436887977340a27bc96c
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-07 11:08:30 -0700

support for debug prints


---

diff --git a/src/preload.c b/src/preload.c
index 749857e..55ee535 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -48,6 +48,7 @@
 #include <netinet/tcp.h>
 #include <unistd.h>
 #include <semaphore.h>
+#include <stdio.h>
 
 #include <rdma/rdma_cma.h>
 #include <rdma/rdma_verbs.h>
@@ -98,6 +99,8 @@ static int rq_size;
 static int sq_inline;
 static int fork_support;
 
+static FILE fout;
+
 enum fd_type {
 	fd_normal,
 	fd_rsocket,
@@ -286,6 +289,7 @@ static void init_preload(void)
 	rs.getsockopt = dlsym(RTLD_DEFAULT, "rgetsockopt");
 	rs.fcntl = dlsym(RTLD_DEFAULT, "rfcntl");
 
+	fout = fopen("rs-out.txt", "w+");
 	getenv_options();
 	init = 1;
 out:
@@ -967,12 +971,16 @@ int fstat(int socket, struct stat *buf)
 {
 	int fd, ret;
 
+	fprintf(fout, "fstat");
 	if (fd_get(socket, &fd) == fd_rsocket) {
 		ret = real.fstat(socket, buf);
+		fprintf(fout, "fstat - rsocket %d", ret);
 		if (!ret)
 			buf->st_mode = (buf->st_mode & ~S_IFMT) | __S_IFSOCK;
 	} else {
 		ret = real.fstat(fd, buf);
+		fprintf(fout, "fstat - normal %d", ret);
 	}
+	fflush(fout);
 	return ret;
 }
